addons/website/tests/test_iap.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import odoo
+import odoo.tests
+from odoo.addons.iap.tools import iap_tools
+
+@odoo.tests.tagged('external')
+class TestIap(odoo.tests.HttpCase):
+
+    def test_01_industries_lang(self):
+        """Ensure that the industries are translated in all the languages
+        supported by the configurator."""
+        with iap_tools.enabled_iap_test():
+            configurator_languages = ['en', 'de', 'es', 'fr', 'hr', 'hu', 'id', 'it', 'mk', 'nl', 'pt', 'ru', 'zh']
+            industries_by_lang = {}
+            for lang in configurator_languages:
+                industries = self.env['website']._website_api_rpc('/api/website/1/configurator/industries', {'lang': lang})['industries']
+                industries_by_lang[lang] = {industry['id']: industry['label'] for industry in industries}
+
+            for (index, lang1) in enumerate(configurator_languages):
+                for lang2 in configurator_languages[index + 1:]:
+                    has_diff = False
+                    self.assertEqual(len(industries_by_lang[lang1]), len(industries_by_lang[lang2]), "Different number of industries between %s and %s" % (lang1, lang2))
+                    for id in industries_by_lang[lang1]:
```

Comment:
That many `for` is a red flag, both the fact that we loop 2 times on lang and the fact that there is 3 nested `for`.

Unless I am missing something, it doesn't really makes sense, why are we doing this kind of loop?
```
lang1 vs lang2
lang1 vs lang3
lang1 vs lang4
lang1 vs lang5

lang2 vs lang3
lang2 vs lang4
lang2 vs lang5

lang3 vs lang4
lang4 vs lang5

lang4 vs lang5
```

Isn't
```
lang1 vs lang2
lang1 vs lang3
lang1 vs lang4
lang1 vs lang5
```
Enough? If a lang is translated, its enough to compare it to english/first lang only, it's impossible that the lang is translated the same for every terms as another lang.

This whole test can probably just become:
```py
            def _get_industries(lang):
                industries = self.env['website']._website_api_rpc('/api/website/1/configurator/industries', {'lang': lang})['industries']
                return {industry['id']: industry['label'] for industry in industries}

            english_terms = _get_industries('en')
            for lang in ['de', 'es', 'fr', 'hr', 'hu', 'id', 'it', 'mk', 'nl', 'pt', 'ru', 'zh']:
                translated_terms = _get_industries(lang)
                has_diff = False
                self.assertEqual(len(english_terms), len(translated_terms), "Different number of industries between 'en' and %s" % lang)
                for id, english_label in english_terms.items():
                    translated_label = translated_terms.get(id, False)
                    self.assertTrue(translated_label, "Industry %s is not in %s" % (english_label, lang))
                    if english_label != translated_label:
                        has_diff = True
                        break
                self.assertTrue(has_diff, "No difference found between 'en' and %s" % lang)
```
With only 2 `for`.
It can probably also be optimized with the usage of `zip` if we assume this is not possible to have a different structure (IDs) for the translated call.

Reply:
Indeed I wanted to be sure that lang2 is different of lang3 but as you think it is not necessary to check, I will simplify the test :+1: